rir/src/runtime/Code.h

```diff
@@ -98,11 +101,33 @@ struct Code {
         unsigned srcIdx;
     };
 
+    SEXP container() {
```

Comment:
would it make sense to have a super-class for Runtime objects (ie. for dispatchtable, code, function). Something like:

```
tamplate<typename BASE>
struct RirRuntimeObject<BASE, PTRS> {
  rir::rir_header info;
  RirRuntimeObject(SEXP* gc_area_start, size_t gc_area_length) {
    memset(gc_area_start, ....)_
    initialize_rir_header ....
  }
  setElement(size_t pos, SEXP v) {
    EXTERNALSXP_SET_ENTRY(...)
  }
  SEXP container();
  BASE* unpack(SEXP);
  BASE* check(SEXP);
};
```

Reply:
Could then also get rid of the header and put the stuff here. Also, the last two functions are static

Reply:
That's a good idea. The repetition was starting to annoy me.

I don't know how likely it is that we'll add another runtime object, but the (non-C++) [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) would apply.